llvm/lib/Target/X86/X86InstrRAOINT.td

```diff
@@ -25,21 +25,30 @@ def X86rao_xor  : SDNode<"X86ISD::AXOR", SDTRAOBinaryArith,
 def X86rao_and  : SDNode<"X86ISD::AAND", SDTRAOBinaryArith,
                          [SDNPHasChain, SDNPMayStore, SDNPMemOperand]>;
 
-multiclass RAOINT_BASE<string OpcodeStr> {
+multiclass RAOINT_BASE<string m, string suffix = ""> {
   let Predicates = [HasRAOINT] in
-    def 32mr : I<0xfc, MRMDestMem, (outs), (ins i32mem:$dst, GR32:$src),
-                 !strconcat("a", OpcodeStr, "{l}\t{$src, $dst|$dst, $src}"),
-                 [(!cast<SDNode>("X86rao_" # OpcodeStr) addr:$dst, GR32:$src)]>,
-               Sched<[WriteALURMW]>;
+    def 32mr#suffix : I<0xfc, MRMDestMem, (outs), (ins i32mem:$src1, GR32:$src2),
+                        !strconcat("a", m, "{l}\t", binop_args),
```

Comment:
Use `BinOpMR_M`. The schedule should be corrected to
`WriteALURMW, ReadDefault, ReadDefault, ReadDefault, ReadDefault, ReadDefault`. You can create a separate PR for this.

Reply:
What's the difference between [WriteALURMW] and [WriteALURMW, ReadDefault, ReadDefault, ReadDefault, ReadDefault, ReadDefault]

Reply:
You will find the difference if you check the instruction in MCA test